### PR TITLE
[wip] Always persist service status update

### PR DESCRIPTION
**Pull request by bprashanth**
Haven't properly tested this yet.

### CODE REVIEW

File: pkg/controller/service/servicecontroller.go [hunk 1]
```diff
@@ -295,14 +295,8 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
...
 
-	// Write the state if changed
-	// TODO: Be careful here ... what if there were other changes to the service?
-	if !v1.LoadBalancerStatusEqual(previousState, &service.Status.LoadBalancer) {
-		if err := s.persistUpdate(service); err != nil {
-			return fmt.Errorf("Failed to persist updated status to apiserver, even after retries. Giving up: %v", err), notRetryable
-		}
-	} else {
-		glog.V(2).Infof("Not persisting unchanged LoadBalancerStatus to registry.")
+	if err := s.persistUpdate(service); err != nil {
```
Comment (thockin) [inline pkg/controller/service/servicecontroller.go]: I would include a comment here explaining why we do this even if it *seems* to not be needed.

**Comment by thockin** [comment]
This LGTM overall

**Comment by freehan** [comment]
Can you include the log level bump from https://github.com/kubernetes/kubernetes/pull/38419?

**Comment by bprashanth** [comment]
Lets get the status pr in seperately so we don't end up reverting the much needed logging

**Pull request closed without merging**